consensus/subscribers.go

```diff
@@ -0,0 +1,86 @@
+package consensus
+
+// A Subscriber is an object that receives updates to the consensus set every
+// time there is a change in consensus.
+type Subscriber interface {
+	// ReceiveConsensusUpdate notifies subscribers that there has been a change in the
+	// consensus set, and provides an enumeration of the changes.
+	ReceiveConsensusUpdate(revertedBlocks []Block, appliedBlocks []Block)
+}
```

Comment:
docstring is misleading; the method isn't the one doing the notification, it's the one processing the changes. `ProcessConsensusUpdate` might be a better name.

Also, doesn't `BlockSince` just return the `BlockID`s? Seems like that could be more efficient, esp. if the Subscriber doesn't need the contents of every block.